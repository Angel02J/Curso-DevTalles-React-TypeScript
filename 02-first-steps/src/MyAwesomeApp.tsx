import type { CSSProperties } from "react";

const firstName = 'Jose';
const lastName = 'Lugo';

const favoriteGames = ['Elden Ring', 'Smash', 'Metal Gears'];
const isActive = true;

const address = {
    zipCode: 'ABC-123',
    country: 'Canada'
}

const myStyles: CSSProperties = {
    backgroundColor: '#fafafa',
    borderRadius: 20,
    padding: 10,
    marginTop: 30,
}

const MyAwesomeApp = () => {



    return (
        <>
            <h1 data-testid="first-name-title">{firstName}</h1>
            <h3>{lastName}</h3>

            <p>{favoriteGames.join(', ')}</p>

            <h1>{isActive ? 'Activo' : 'No activo'}</h1>

            <p
                style={myStyles}
            >
                {JSON.stringify(address)}
            </p>
        </>
    )
}

export default MyAwesomeApp