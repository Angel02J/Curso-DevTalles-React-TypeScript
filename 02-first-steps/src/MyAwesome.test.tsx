import { describe, expect, test } from "vitest";
import { render, screen } from '@testing-library/react';
import MyAwesomeApp from "./MyAwesomeApp";


describe('MyAwesomeApp', () => {
    test('Should render firstName and lastName', () => {

        const { container } = render(<MyAwesomeApp />);

        // screen.debug();

        const h1 = container.querySelector('h1');
        expect(h1?.innerHTML).toContain('Jose');

        const h3 = container.querySelector('h3');
        expect(h3?.innerHTML).toContain('Lugo');


    });

    test('Should render firstName and lastName - screen', () => {

        render(<MyAwesomeApp />);

        screen.debug();

        // const h1 = screen.getByRole('heading', {
        //     level: 1
        // });

        // console.log(h1.innerHTML);

        const h1 = screen.getByTestId('first-name-title');
        expect(h1.innerHTML).toContain('Jose');
    });
});