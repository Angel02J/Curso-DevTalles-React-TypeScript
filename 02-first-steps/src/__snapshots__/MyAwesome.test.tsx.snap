// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > Should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
      Jose
    </h1>
    <h3>
      Lugo
    </h3>
    <p>
      4
    </p>
    <p
      class="mi-clase-favorita"
    >
      Elden Ring, Smash, Metal Gears
    </p>
    <h1>
      Activo
    </h1>
    <p
      style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
    >
      {"zipCode":"ABC-123","country":"Canada"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > Should match snapshot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
    Jose
  </h1>
  <h3>
    Lugo
  </h3>
  <p>
    4
  </p>
  <p
    class="mi-clase-favorita"
  >
    Elden Ring, Smash, Metal Gears
  </p>
  <h1>
    Activo
  </h1>
  <p
    style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
  >
    {"zipCode":"ABC-123","country":"Canada"}
  </p>
</div>
`;
